### PR TITLE
Enhance library functions for tools.

### PR DESCRIPTION
**Pull request by jmchilton**
- Enhance lint and loaders to be able to handle tool parser objects instead of just raw XML. This will allow non-Galaxy-XML tool linting (e.g. CWL) and allow linters to use higher-level abstraction.
- Replace tol-level XML linter with a "general" linter that parses attributes using the ToolSource abstraction.
- Small updates to linting for recent additions to tool framework.
- More docstring usage and improved import order.
- Fixes discovered through planemo for "finding" non-XML tools.

### CODE REVIEW

**Comment by jmchilton** [comment]
Awesome - thanks a bunch for the merge and spelling fixes @dannon.

**Comment by dannon** [comment]
Anytime!

File: lib/galaxy/tools/lint.py [hunk 1]
```diff
@@ -1,29 +1,60 @@
+"""This modules contains the functions that driving the tool linting framework."""
```
Comment (nsoranzo) [inline lib/galaxy/tools/lint.py:1]: "that driving": "that" should be removed on next PR.

Comment (dannon) [inline lib/galaxy/tools/lint.py:1]: Easy enough to do it now, PRs are cheap.  #2289

**Pull request merged**